### PR TITLE
Multiple hosts support, try n2

### LINKED ISSUE #1470: PostgreSQL 10 support
PostgreSQL v10 was released on Oct 05, 2017

---

One of the most interesting feature that affects `node-postgres` is [multiple host names or addresses in libpq connection strings and URIs](https://www.postgresql.org/docs/10/static/libpq-connect.html#libpq-connect-host).

This feature allows automatic support for replicas on the connection level, i.e. the first connection is tried, and if fails - the next one, and so on.

[From documentation](https://www.postgresql.org/docs/10/static/libpq-connect.html#libpq-multiple-hosts):

> It is possible to specify multiple hosts to connect to, so that they are tried in the given order. In the Keyword/Value format, the host, hostaddr, and port options accept a comma-separated list of values. The same number of elements must be given in each option, such that e.g. the first hostaddr corresponds to the first host name, the second hostaddr corresponds to the second host name, and so forth. As an exception, if only one port is specified, it applies to all the hosts.

### PR DESCRIPTION
**Pull request by zxqfox**
Closes #1470

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,16 +20,16 @@
   "main": "./lib",
   "dependencies": {
     "buffer-writer": "2.0.0",
+    "connection-string": "^3.0.4",
     "packet-reader": "1.0.0",
     "pg-connection-string": "0.1.3",
+    "pg-native": "^3.0.0",
```
Comment (charmander) [inline package.json]: pg isn’t supposed to have a dependency on pg-native, since pg-native is optional.

File: package.json [hunk 2]
```diff
@@ -20,16 +20,16 @@
   "main": "./lib",
   "dependencies": {
     "buffer-writer": "2.0.0",
+    "connection-string": "^3.0.4",
```
Comment (charmander) [inline package.json]: connection-string follows a different spec from libpq. I think a PostgreSQL driver should stick with the libpq format.

Comment (zxqfox) [reply package.json]: I know.

**Comment by zxqfox** [comment]
Oh, not sure I can finish this in the near future. Sorry ;-(

Todo:
- [ ] `connection-string` drop (or persuade to make a BREAKING change)
- [ ] JS-version of server lookup with attributes `target_session_attribute=read-write` (analogue to libpq build-in)
- [ ] drop `pg-native` dep

Still not sure what is actually right to do with `connection-string`. Fellows in `pg-connection-string` tells me to replace it with `connection-string` here, and here you saying we need to make it work with non-valid spec
Prob we should drop old behaviour because it disrespect modern URL/connstring specs

**Pull request closed without merging**